### PR TITLE
[Enhancement] Optimize materialized view refresh task by passing properties to task

### LINKED ISSUE #8859: [Enhancement] materialized view refresh task fails because of query timeout
## Enhancement
Materialized view refresh task fails because of query timeout. The reason is that the query_timeout session variables can not be passed to task.
So I consider to optimize it by passing the session variables when define materialized view with hints, like followings:

CREATE MATERIALIZED VIEW mv1_partition_by_column
PARTITION BY k1
distributed by hash(k2)
refresh async START('2122-12-31') EVERY(INTERVAL 1 HOUR)
PROPERTIES('replication_num' = '1')
as
select /*+ SET_VAR(query_timeout = 1000) */ k1, k2 from tbl1;

query_timeout can be passed to task through SET_VAR hints

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8859 

## Problem Summary(Required) ：

Optimize materialized view refresh task by passing properties to task through hints

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|16|16|100.00%|[]|

**Comment by ABingHuang** [comment]
run starrocks_be_unittest

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|16|16|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -2913,6 +2919,20 @@ public void createMaterializedView(CreateMaterializedViewStatement stmt)
...
             throw new DdlException(e.getMessage());
         }
+        // validate optHints
+        Map<String, String> optHints = null;
+        QueryRelation queryRelation = stmt.getQueryStatement().getQueryRelation();
+        if (queryRelation instanceof SelectRelation) {
+            SelectRelation selectRelation = (SelectRelation) queryRelation;
+            optHints = selectRelation.getSelectList().getOptHints();
+            if (optHints != null && !optHints.isEmpty()) {
+                SessionVariable sessionVariable = VariableMgr.newSessionVariable();
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2929]: could you get the ConnectContext here? if could, we could get the SessionVariable from the ConnectContext.

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2929]: optHints here are passed to task, not used by current ConnectContext. And this SessionVariable is used to checked the validity of optHints, so I use a new temp SessionVariable here.

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_be_build

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Pull request merged**